### DISCUSSION TITLE
10.5 to 10.6 migration with intention to migrate to 11

### DISCUSSION THREAD

**Discussion by lohitpereira** [pimcore/pimcore#16116] [Q&A / Support]
I  set `"pimcore/pimcore": "10.6"` in composer.json and  executed `composer update -W`. here is a screen shot of the following error that I am unable to fix. Here is the path to the file mentioned `vendor\pimcore\pimcore\lib\Model\Dao\AbstractDao.php` Could someone help me with this. 
![composer](https://github.com/pimcore/pimcore/assets/115552460/f954bf6a-c303-462b-8d81-01d2029b307b)

**Comment by Thejaswini-Rao-U** [comment]
@lohitpereira ,

previous versions are migrated successfully ? like 6.9 and 10 version ?

because i could see its issue of ur existing db columns, might be old db columns are not migrated successfully to 6.9 version which is needed for 10.6 version ? this could be the issue.

**Reply by lohitpereira** [reply]
I installed v10.5, initialized pimcore with the necessary assets and other data. Later I imported the objects from the csv. 
Basically I started from a fresh install of v10.5

I executed the command from https://github.com/pimcore/pimcore/discussions/15890#discussioncomment-6901927
and here is the output.

`[warning] Migration Pimcore\Bundle\CoreBundle\Migrations\Version20230508121105 was executed but did not result in any SQL statements.
[notice] finished in 45550.6ms, used 78M memory, 6 migrations executed, 6 sql queries`

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira , if u have installed from skeleton project which is given in the pimcore github, then please follow the steps given in the pimcore document.

as you have old version db , you need to migrate everything to the newer version as per required. otherwise it might get issues when you migrate.

**Reply by lohitpereira** [reply]
I just executed the following code

`C:\wamp64\bin\php\php8.0.26\php.exe --php-ini=C:\wamp64\bin\php\php8.0.26\php.ini bin/console --ansi assets:install --symlink --relative --ignore-maintenance-mode public`

post doctrine migration and there was no error. 


**Successfully migrated** to v10.6 & now heading towards v11

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira , Okay Great!

**Reply by lohitpereira** [reply]
As per the upgrade notes, Admin Bundle has been moved to `pimcore/admin-ui-classic-bundle ` while I use use `Pimcore\Bundle\AdminBundle\Controller\AdminController` I am unable to find AdminController in `pimcore/admin-ui-classic-bundle`.  would `vendor\pimcore\admin-ui-classic-bundle\src\Controller\GDPR\AdminController.php` be relevant ?

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira , can you share where you want to use admin controller?

**Reply by lohitpereira** [reply]
my Admin Controller is at `\src\Controller\Admin\AdminController.php` 
I added `use Pimcore\Bundle\AdminBundle\Controller\GDPR\AdminController` and the error is gone .  had missed out `Controller` in the syntax.

**Reply by lohitpereira** [reply]
currently I Am stuck at this point 
![image](https://github.com/pimcore/pimcore/assets/115552460/9492b3b1-b753-428d-9e38-935e9716458e)

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira 
please read pimcore document.. its mentioned here use **UserAwareController**

https://pimcore.com/docs/pimcore/10.6/Development_Documentation/Installation_and_Upgrade/Upgrade_Notes/index.html

**Reply by lohitpereira** [reply]
Hi @Thejaswini-Rao-U , I have implemented the changes as mentioned in the upgrade notes and have resolved a number of issues.

At the moment I am stuck at this point `security.command.debug_firewall` I do not have this configuration in my security.yml as I do not use it. I have attached a screenshot below. How do I get past this ?

![image](https://github.com/pimcore/pimcore/assets/115552460/1364020b-e485-4955-bae3-df11526ddfcf)

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira , can you share your security.yaml file details here?

**Reply by lohitpereira** [reply]
Yes @Thejaswini-Rao-U ,  Here it is.

`security:
    enable_authenticator_manager: true

    providers:
        pimcore_admin:
            id: Pimcore\Security\User\UserProvider

    firewalls:
        dev:
            pattern: ^/(_(profiler|wdt)|css|images|js)/
            security: false

        pimcore_admin_webdav:
            pattern: ^/admin/asset/webdav
            provider: pimcore_admin
            http_basic: ~

        pimcore_admin:
            pattern: ^/admin(/.*)?$
            stateless: true
            provider: pimcore_admin
            login_throttling:
                max_attempts: 3
                interval: '5 minutes'

            logout:
                path: pimcore_admin_logout
                target: pimcore_admin_login
            custom_authenticators:
                - Pimcore\Bundle\AdminBundle\Security\Authenticator\AdminTokenAuthenticator
            two_factor:
                auth_form_path: /admin/login/2fa                   # Path or route name of the two-factor form
                check_path: /admin/login/2fa-verify                # Path or route name of the two-factor code check
                default_target_path: /admin            # Where to redirect by default after successful authentication
                always_use_default_target_path: false  # If it should always redirect to default_target_path
                auth_code_parameter_name: _auth_code   # Name of the parameter for the two-factor authentication code
                trusted_parameter_name: _trusted       # Name of the parameter for the trusted device option
                multi_factor: false                    # If ALL active two-factor methods need to be fulfilled (multi-factor authentication)

    access_control:
         - { path: ^/admin/settings/display-custom-logo, roles: PUBLIC_ACCESS }
        - { path: ^/admin/login/2fa-verify, roles: IS_AUTHENTICATED_2FA_IN_PROGRESS}
        - { path: ^/admin/login/2fa, roles: IS_AUTHENTICATED_2FA_IN_PROGRESS}
        - { path: ^/admin/login$, roles: PUBLIC_ACCESS }
        - { path: ^/admin/login/(login|lostpassword|deeplink|csrf-token)$, roles: PUBLIC_ACCESS }
        - { path: ^/admin, roles: ROLE_PIMCORE_USER }

    role_hierarchy:
        # Pimcore admin  // DO NOT CHANGE!
        ROLE_PIMCORE_ADMIN: [ROLE_PIMCORE_USER]

`

**Reply by lohitpereira** [reply]
set `security.firewalls.pimcore_admin: '%pimcore_admin_bundle.firewall_settings%' `

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira , i was busy could not get time to check. anyhow thats the correct answer. it supposed to be. Its great u fixed it

**Reply by lohitpereira** [reply]
Thanks for the help this far.
At the moment the page after the login screen does not load. 

In the logsI found `security.DEBUG: Authenticator does not support the request` 
for all the 3 authenticators namely AdminTokenAuthenticator, TwoFactorAuthenticator & FormLoginAuthenticator

I used userName/password to login on my local.

**Reply by Thejaswini-Rao-U** [reply]
@lohitpereira, below code  isn't works for the login page load ?


        # Pimcore admin form login // DO NOT CHANGE!
        pimcore_admin:
            pattern: ^/admin(/.*)?$
            # admin firewall is stateless as we open the admin
            # session on demand for non-blocking parallel requests
            stateless: true
            provider: pimcore_admin
            login_throttling:
                max_attempts: 3
                interval: '5 minutes'
            logout:
                path: pimcore_admin_logout
                target: pimcore_admin_login
            custom_authenticators:
                    - Pimcore\Bundle\AdminBundle\Security\Authenticator\AdminLoginAuthenticator
                    - Pimcore\Bundle\AdminBundle\Security\Authenticator\AdminTokenAuthenticator
                    - Pimcore\Bundle\AdminBundle\Security\Authenticator\AdminSessionAuthenticator
            two_factor:
                auth_form_path: /admin/login/2fa                   # Path or route name of the two-factor form
                check_path: /admin/login/2fa-verify                # Path or route name of the two-factor code check
                default_target_path: /admin            # Where to redirect by default after successful authentication
                always_use_default_target_path: false  # If it should always redirect to default_target_path
                auth_code_parameter_name: _auth_code   # Name of the parameter for the two-factor authentication code
                trusted_parameter_name: _trusted       # Name of the parameter for the trusted device option
                multi_factor: false

**Reply by lohitpereira** [reply]
Hi @Thejaswini-Rao-U ,

Had to clear Browser cache. and the page after login loaded.
Now I do not see the Documents, assets and Data Objects.

looking through the network calls I found that a call to  
`http://localhost/admin/user/get-current-user?_dc=...` returns a 500 error response.

looking into the dev.log and php.log, here is what I found.

**\var\log\dev.log**
`admin_statistics.INFO: App\Controller\Admin\UserController::getCurrentUserAction [2,"pimcore_admin_user_getcurrentuser",[],[]] []`

`php.CRITICAL: Uncaught Error: Pimcore\Model\DataObject\Concrete::classId(): Return value must be of type string, null returned {"exception":"[object] (TypeError(code: 0): Pimcore\\Model\\DataObject\\Concrete::classId(): Return value must be of type string, null returned at C:\\wamp64\\www\\pomodoro\\pomodoro\\vendor\\pimcore\\pimcore\\models\\DataObject\\Concrete.php:114)"} []`

`request.CRITICAL: Uncaught PHP Exception TypeError: "Pimcore\Model\DataObject\Concrete::classId(): Return value must be of type string, null returned" at C:\wamp64\www\pomodoro\pomodoro\vendor\pimcore\pimcore\models\DataObject\Concrete.php line 114 {"exception":"[object] (TypeError(code: 0): Pimcore\\Model\\DataObject\\Concrete::classId(): Return value must be of type string, null returned at C:\\wamp64\\www\\pomodoro\\pomodoro\\vendor\\pimcore\\pimcore\\models\\DataObject\\Concrete.php:114)"} []`

**\var\log\php.log**
`php.CRITICAL: Uncaught Error: Call to a member function getFieldDefinitions() on null {"exception":"[object] (Error(code: 0): Call to a member function getFieldDefinitions() on null at C:\\wamp64\\www\\pomodoro\\pomodoro\\vendor\\pimcore\\pimcore\\models\\DataObject\\Concrete.php:544)"} []`

`request.CRITICAL: Uncaught PHP Exception Error: "Call to a member function getFieldDefinitions() on null" at C:\wamp64\www\pomodoro\pomodoro\vendor\pimcore\pimcore\models\DataObject\Concrete.php line 544 {"exception":"[object] (Error(code: 0): Call to a member function getFieldDefinitions() on null at C:\\wamp64\\www\\pomodoro\\pomodoro\\vendor\\pimcore\\pimcore\\models\\DataObject\\Concrete.php:544)"} []`

`php.CRITICAL: Uncaught Error: Pimcore\Model\DataObject\Concrete::classId(): Return value must be of type string, null returned {"exception":"[object] (TypeError(code: 0): Pimcore\\Model\\DataObject\\Concrete::classId(): Return value must be of type string, null returned at C:\\wamp64\\www\\pomodoro\\pomodoro\\vendor\\pimcore\\pimcore\\models\\DataObject\\Concrete.php:114)"} []`
